src/Components/Aspire.MongoDB.Driver/AspireMongoDbDriverExtensions.cs

```diff
@@ -0,0 +1,218 @@
+    /// <param name="name">The name of the component, which is used as the <see cref="ServiceDescriptor.ServiceKey"/> of the service and also to retrieve the connection string from the ConnectionStrings configuration section.</param>
+    /// <param name="configureSettings">An optional delegate that can be used for customizing options. It's invoked after the settings are read from the configuration.</param>
+    /// <remarks>Reads the configuration from "Aspire:MongoDB:Driver:{name}" section.</remarks>
+    /// <param name="configureClientSettings">An optional delegate that can be used for customizing MongoClientSettings.</param>
+    /// <exception cref="ArgumentNullException">Thrown if mandatory <paramref name="builder"/> is null.</exception>
+    /// <exception cref="InvalidOperationException">Thrown when mandatory <see cref="MongoDbSettings.ConnectionString"/> is not provided.</exception>
+    public static void AddKeyedMongoDbDataSource(
+        this IHostApplicationBuilder builder,
+        string name,
+        Action<MongoDbSettings>? configureSettings = null,
+        Action<MongoClientSettings>? configureClientSettings = null)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(name);
+
+        builder.AddMongoDbDataSource(
+            $"{DefaultConfigSectionName}:{name}",
+            configureSettings,
+            configureClientSettings,
+            connectionName: name,
+            serviceKey: name);
+    }
+
+    private static void AddMongoDbDataSource(
+        this IHostApplicationBuilder builder,
+        string configurationSectionName,
+        Action<MongoDbSettings>? configureSettings,
+        Action<MongoClientSettings>? configureClientSettings,
+        string connectionName,
+        string? serviceKey)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        var settings = builder.GetMongoDbSettings(
+            connectionName,
+            configurationSectionName,
+            configureSettings);
+
+        var mongoUrl = MongoUrl.Create(settings.ConnectionString);
+
+        builder.AddMongoClient(
+            settings,
+            connectionName,
+            configurationSectionName,
+            configureClientSettings,
+            serviceKey);
+
+        if (!string.IsNullOrWhiteSpace(mongoUrl.DatabaseName))
+        {
+            builder.AddMongoDatabase(mongoUrl, serviceKey);
+        }
+
+        if (settings.HealthCheckEnabled)
+        {
+            builder.AddHealthCheck(
+                mongoUrl,
+                connectionName,
+                serviceKey,
+                settings.HealthCheckTimeout);
+        }
+    }
+
+    private static IServiceCollection AddMongoClient(
+        this IHostApplicationBuilder builder,
+        MongoDbSettings mongoDbSettings,
+        string connectionName,
+        string configurationSectionName,
+        Action<MongoClientSettings>? configureClientSettings = null,
+        string? serviceKey = null)
```

Comment:
(nit) these are always supplied.

```suggestion
        Action<MongoClientSettings>? configureClientSettings,
        string? serviceKey)
```